sync-core/src/main/java/com/cloudant/sync/query/QueryExecutor.java

```diff
@@ -205,7 +227,9 @@ private boolean validateSortDocument(List<Map<String, String>> sortDocument) {
                 if (accumulator == null) {
                     accumulator = new HashSet<String>(childIds);
                 } else {
-                    accumulator = Sets.intersection(accumulator, childIds);
+                    if (childIds != null) {
+                        accumulator = Sets.intersection(accumulator, childIds);
+                    }
```

Comment:
Feels more logical to me to have an `if (childIds == null) { continue; }` after the assignment to `childIds`, as this avoids creating the accumulator if there are never any children. Also, perhaps the `HashSet` constructor might not like nil?